src/Neo/SmartContract/JsonSerializer.cs

```diff
@@ -191,8 +191,7 @@ private static StackItem Deserialize(JToken json, ref uint maxStackSize, Referen
                 case JNumber num:
                     {
                         if ((num.Value % 1) != 0) throw new FormatException("Decimal value is not allowed");
```

Comment:
And BTW, this line saves us from things like `100.123e+2`. The exception is thrown on attempt to parse it, thus, the second problem described in #2879 isn't relevant:
```
Error Message:
   Test method Neo.UnitTests.SmartContract.UT_JsonSerializer.Deserialize_Array_Bool_Str_Num threw exception: 
System.FormatException: Decimal value is not allowed
  Stack Trace:
      at Neo.SmartContract.JsonSerializer.Deserialize(JToken json, UInt32& maxStackSize, ReferenceCounter referenceCounter) in /home/anna/Documents/GitProjects/neo-project/neo/src/Neo/SmartContract/JsonSerializer.cs:line 193
   at Neo.SmartContract.JsonSerializer.Deserialize(JToken json, UInt32& maxStackSize, ReferenceCounter referenceCounter) in /home/anna/Documents/GitProjects/neo-project/neo/src/Neo/SmartContract/JsonSerializer.cs:line 184
   at Neo.SmartContract.JsonSerializer.Deserialize(JToken json, ExecutionEngineLimits limits, ReferenceCounter referenceCounter) in /home/anna/Documents/GitProjects/neo-project/neo/src/Neo/SmartContract/JsonSerializer.cs:line 168
   at Neo.UnitTests.SmartContract.UT_JsonSerializer.Deserialize_Array_Bool_Str_Num() in /home/anna/Documents/GitProjects/neo-project/neo/tests/Neo.UnitTests/SmartContract/UT_JsonSerializer.cs:line 290
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
```